finagle-http2/src/main/scala/com/twitter/finagle/http2/Http2Transporter.scala

```diff
@@ -94,67 +137,38 @@ private[http2] object Http2Transporter {
             case req => Future.exception(new IllegalArgumentException(
               s"expected a Netty4Response, got a ${req.getClass.getName}"))
           }
-        }
       }
     }
 
-    private[this] def unsafeCast(trans: Transport[HttpObject, HttpObject]): Transport[Any, Any] =
-      trans.map(_.asInstanceOf[HttpObject], _.asInstanceOf[Any])
-
-    def apply(addr: SocketAddress): Future[Transport[Any, Any]] = Option(map.get(addr)) match {
+  @scala.annotation.tailrec
+  final def apply(addr: SocketAddress): Future[Transport[Any, Any]] =
+    Option(transporterCache.get(addr)) match {
       case Some(f) =>
+        println(s"aleady cached $addr")
         f.flatMap { multiplexed =>
-          multiplexed(addr).map(unsafeCast _)
```

Comment:
ffti: I like to make eta expansion explicit so that I feel bad about the allocation, but it's your call.